### ISSUE TITLE
GetDevices call crashes with "panic: Cannot unmarshal field with type 'types.DateTime'"

### ISSUE THREAD

**Issue by uhppoted** [uhppoted/uhppoted-rest#3]
```
uhppoted-rest -config "C:\\ProgramData\\uhppoted\\uhppoted.conf" -console -debug

2021/04/08 09:58:01 uhppoted-rest daemon v0.6.x - Microsoft Windows (PID 9124)
2021/04/08 09:58:01 ... listening on port 8080

... request
...          00000000  17 94 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...          00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...          00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...          00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...
... sent 64 bytes to 192.168.1.255:60000
... received 64 bytes from 192.168.1.131:60000
...          00000000  17 94 00 00 e5 d8 4f 0d  c0 a8 01 8d ff ff ff 00  |......O.........|
...          00000010  c0 a8 01 01 00 66 0d 4f  d8 e5 08 98 20 19 05 21  |.....f.O.... ..!|
...          00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...          00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...
... request
...          00000000  17 20 00 00 e5 d8 4f 0d  00 00 00 00 00 00 00 00  |. ....O.........|
...          00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...          00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...          00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
...
... sent 64 bytes to 192.168.1.255:60000
... received 64 bytes from 192.168.1.131:60000
... response
...          00000000  17 20 00 00 e5 d8 4f 0d  00 00 00 00 00 00 00 00  |. ....O.........|
...          00000010  00 00 00 00 00 00 00 00  00 00 00 00 01 01 00 00  |................|
...          00000020  00 00 00 00 00 16 03 34  00 00 00 00 00 00 00 00  |.......4........|
...          00000030  00 00 00 21 04 08 00 00  00 00 00 00 00 00 00 00  |...!............|
...
panic: Cannot unmarshal field with type 'types.DateTime'

goroutine 8 [running]:
github.com/uhppoted/uhppote-core/encoding/UTO311-L0x.unmarshal(0xc00016d800, 0x40, 0x800, 0xb78640, 0xc000076300, 0x199, 0x1c1, 0x380)
       .../github.com/uhppoted/uhppote-core@v0.6.11-0.20210406214223-dfa6e21e6b7b/encoding/UTO311-L0x/UT0311-L0x.go:297 +0x1505
       .../github.com/uhppoted/uhppote-core/encoding/UTO311-L0x.Unmarshal(0xc00016d800, 0x40, 0x800, 0xaf0160, 0xc000076300, 0x3, 0x3)
       .../github.com/uhppoted/uhppote-core@v0.6.11-0.20210406214223-dfa6e21e6b7b/encoding/UTO311-L0x/UT0311-L0x.go:159 +0x1de
       .../github.com/uhppoted/uhppote-core/uhppote.(*UHPPOTE).receive(0xc000117c20, 0xc000006120, 0xaf0160, 0xc000076300, 0xc0000177c8, 0x0)
       .../github.com/uhppoted/uhppote-core@v0.6.11-0.20210406214223-dfa6e21e6b7b/uhppote/uhppote.go:292 +0x350
       .../github.com/uhppoted/uhppote-core/uhppote.(*UHPPOTE).Execute(0xc000117c20, 0xc00d4fd8e5, 0xb32f40, 0xc0000177c8, 0xaf0160, 0xc000076300, 0x0, 0x0)
       .../github.com/uhppoted/uhppote-core@v0.6.11-0.20210406214223-dfa6e21e6b7b/uhppote/uhppote.go:154 +0x1e7
       .../github.com/uhppoted/uhppote-core/uhppote.(*UHPPOTE).GetStatus(0xc000117c20, 0xc00d4fd8e5, 0xc0001a3d88, 0xc0001a3d20, 0x1)
       .../github.com/uhppoted/uhppote-core@v0.6.11-0.20210406214223-dfa6e21e6b7b/uhppote/get_status.go:16 +0xb2
       .../github.com/uhppoted/uhppoted-rest/commands.healthcheck(0xc000117c20, 0xc000076200, 0xc0000463c0)
       .../uhppoted-rest/commands/run.go:247 +0x3e5
       .../github.com/uhppoted/uhppoted-rest/commands.(*Run).listen.func2(0xc000046410, 0xc000117c20, 0xc000076200, 0xc0000463c0)
       .../uhppoted-rest/commands/run.go:196 +0x65
       created by github.com/uhppoted/uhppoted-rest/commands.(*Run).listen
       .../uhppoted-rest/commands/run.go:193 +0x717
```
```
uhppoted.conf

# SYSTEM  THE REST SERVER IP
bind.address = 192.168.1.130:0
broadcast.address = 192.168.1.255:60000
listen.address = 192.168.1.130:60001

# REST
rest.http.enabled = true
rest.http.port = 8080
rest.https.enabled = false
rest.https.port = 8443
rest.CORS.enabled = true
rest.auth.enabled = false
```

**Comment by twystd** [comment]
Firmware v8.98 is returning a get-status response with an invalid event date/time field from a new controller:
```
...          00000000  17 20 00 00 e5 d8 4f 0d  00 00 00 00 00 00 00 00  |. ....O.........|
...          00000010  00 00 00 00 00 00 00 00  00 00 00 00 01 01 00 00  |................|
...          00000020  00 00 00 00 00 16 03 34  00 00 00 00 00 00 00 00  |.......4........|
...          00000030  00 00 00 21 04 08 00 00  00 00 00 00 00 00 00 00  |...!............|
```

This is the correct behaviour since a new controller has no events but differs from firmware v8.94 and below which AFAIK populated the event field with dummy event information.

**Comment by twystd** [comment]
Updates:

1. Validate _bind_, _broadcast_ and _listen_ address ports
2. Discard invalid replies
3. Commonalise `broadcast` and `broadcastTo` implementations

Pushed to `master` - leaving this issue open pending release.

**Comment by uhppoted** [comment]
Fixed in release v0.6.12
